utils_linux.go

```diff
@@ -291,16 +300,20 @@ func (r *runner) run(config *specs.Process) (int, error) {
 			return -1, err
 		}
 	}
-	handler := <-handlerCh
-	status, err := handler.forward(process, tty, detach)
-	if err != nil {
-		r.terminate(process)
+	if r.notifySocket != nil {
+		if err = r.notifySocket.forward(process, detach); err != nil {
+			r.terminate(process)
```

Comment:
In fact, I also want to move these `r.terminate(process)` to defer, and it's safe to move, because we have ignored all errors in this terminate function, I think we should not write them in everywhere. WDYT @kolyshkin @rata

Reply:
Yes, I think all cleanup actions should be in defer, it will be more clear.
https://github.com/opencontainers/runc/pull/4661/files#r2004617978

Reply:
Seems reasonable to move all of them to a defer(). Please do use a different commit.

Reply:
> In fact, I also want to move these r.terminate(process) to defer

@lifubang makes sense